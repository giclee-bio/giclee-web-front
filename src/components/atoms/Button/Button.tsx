import classNames from 'classnames'

export type Props = {
  children: React.ReactNode
  href?: string
  isLink?: boolean
  onClick?: () => void
  type?: 'light' | 'discord' | 'default'
  width?: 'half' | 'default'
} & ({ href: string; isLink: true } | { isLink?: false; onClick: () => void })

const Button: React.FC<Props> = ({ children, href, isLink, onClick, type, width }) => {
  let classes = classNames('h-10', 'rounded', 'text-center', 'font-bold', 'leading-10')

  classes = width === 'half' ? classNames(classes, 'w-32') : classNames(classes, 'w-64')
  switch (type) {
    case 'light':
      classes = classNames(classes, 'bg-white', 'text-base')
      break
    case 'discord':
      classes = classNames(classes, 'bg-discord', 'text-white')
      break
    default:
      classes = classNames(classes, 'bg-stone-800', 'text-white')
  }

  return isLink ? (
    <a className={`inline-block ${classes}`} href={href}>
      {children}
    </a>
  ) : (
    <button className={classes} onClick={onClick}>
      {children}
    </button>
  )
}

export default Button
