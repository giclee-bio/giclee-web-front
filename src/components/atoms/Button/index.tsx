import Button from './Button'

import type { Props } from './Button'

export default Button
export type { Props }
