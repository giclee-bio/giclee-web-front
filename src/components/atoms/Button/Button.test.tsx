import { render } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  it('snapshot link', async () => {
    const { asFragment } = render(
      <Button href='/hoge' isLink={true}>
        children
      </Button>,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('snapshot button', async () => {
    const { asFragment } = render(<Button onClick={jest.fn()}>children</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
