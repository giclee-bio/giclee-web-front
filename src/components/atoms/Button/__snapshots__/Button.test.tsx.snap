// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinneAccordion snapshot button 1`] = `
<DocumentFragment>
  <button
    class="h-10 rounded text-center font-bold leading-10 w-64 bg-stone-800"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`MinneAccordion snapshot link 1`] = `
<DocumentFragment>
  <a
    class="inline-block h-10 rounded text-center font-bold leading-10 w-64 bg-stone-800"
    href="/hoge"
  >
    children
  </a>
</DocumentFragment>
`;
